### PR TITLE
RIA-1963: Contact Details

### PR DESCRIPTION
**Pull request by alexbaeza**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RIA-1963

### Change description ###

Adds Contact Details Page
Adds Validation
Adds field per Validations
Adds correct logic to highlight the right components on validation failure

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```
### Screenshots ###

image

image

image

### CODE REVIEW

File: app/controllers/contact-details.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    if (!hasSelections(request.selections)) {
+      return res.render('contact-details.njk', { data: data, errors: { selections: 0 } });
+    }
+    const validationErrors = validate(request);
+    if (validationErrors != null) {
+      return res.render('contact-details.njk', {
+        data: data,
+        errors: validationErrors,
+        errorList: flattenToErrorList(validationErrors)
```
Comment (t0nin0s) [inline app/controllers/contact-details.ts]: `Object.values(validationErrors)` does the trick as well...

File: app/utils/fields-validations.ts [hunk 2]
```diff
@@ -120,9 +103,80 @@ function nationalityValidation(obj: object) {
...
+/**
+ * Validates a phone number using using joi validation with regex
+ * @param obj containing the property 'text-message-value' to validate the phone number
+ * @return
+ * 'string.empty': if phone number string is empty
+ * 'string.pattern.base': if phone number does not match format
+ * 'null': if no errors were found
+ */
+function phoneValidation(obj: object) {
+  const emailPattern = new RegExp('^(?:0|\\+?44)(?:\\d\\s?){9,10}$');
```
Comment (t0nin0s) [inline app/utils/fields-validations.ts]: emailPattern?

File: views/components/error-components.njk [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{% from "govuk/components/error-summary/macro.njk" import govukErrorSummary %}
+
+{% macro displayErrorSummary(errorList) %}
+    {{ govukErrorSummary({
+        titleText: i18n.validationErrors.errorSummary,
+        errorList: errorList
+    }) }}
+{% endmacro %}
```
Comment (t0nin0s) [inline views/components/error-components.njk]: Do we need this component? Can we just use the govukErrorSummary? We are not adding any logic here

Comment (alexbaeza) [reply views/components/error-components.njk]: You're right I'll remove on next commit

File: app/controllers/contact-details.ts (hunk 1 above)
Comment (alexbaeza) [reply app/controllers/contact-details.ts]: Thanks ! 👍

**Pull request merged**